### PR TITLE
add message in cfg valid condition if override unsupported mgmt state

### PR DESCRIPTION
**Pull request by gabemontero**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1659351

### CODE REVIEW

File: pkg/stub/handler.go [hunk 1]
```diff
@@ -958,6 +958,12 @@ func (h *Handler) ProcessManagementField(srcfg *v1alpha1.SamplesResource) (bool,
 	default:
 		// force it to Managed if they passed in something funky, including the empty string
 		logrus.Warningf("Unknown management state %s specified; switch to Managed", srcfg.Spec.ManagementState)
+		cfgvalid := srcfg.Condition(v1alpha1.ConfigurationValid)
+		cfgvalid.Message = fmt.Sprintf("Unexpected management state %v received, switching to %v", srcfg.Spec.ManagementState, operatorsv1api.Managed)
+		now := kapis.Now()
+		cfgvalid.LastTransitionTime = now
+		cfgvalid.LastUpdateTime = now
+		srcfg.ConditionUpdate(cfgvalid)
```
Comment (bparees) [inline pkg/stub/handler.go]: Thinking about this more, why not:

1) mark configurationvalid=false (with the message indicating that the management state is invalid)
2) leave the management state as whatever they set it to?

Comment (bparees) [reply pkg/stub/handler.go]: nm, I see we went through this discussion in the bug.  I still kinda prefer what i said above but i'm not gonna re-litigate it after signing off on this approach earlier.

Comment (gabemontero) [reply pkg/stub/handler.go]: I explicitly did not mark it false, as config validation will occur after this

IMM, the whole point was not to do 2)

That said, you lgtm'ed ... so perhaps moot

**Pull request merged**